pkg/storage/kubernetes/ipam.go

```diff
@@ -2,31 +2,168 @@ package kubernetes
 
 import (
 	"context"
+	"crypto/rand"
 	"encoding/json"
 	"fmt"
+	"math/big"
 	"net"
 	"strconv"
 	"strings"
-	"sync"
 	"time"
 
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/tools/leaderelection"
-	"k8s.io/client-go/tools/leaderelection/resourcelock"
-
 	"github.com/dougbtv/whereabouts/pkg/allocate"
 	whereaboutsv1alpha1 "github.com/dougbtv/whereabouts/pkg/api/v1alpha1"
 	"github.com/dougbtv/whereabouts/pkg/logging"
 	"github.com/dougbtv/whereabouts/pkg/storage"
 	whereaboutstypes "github.com/dougbtv/whereabouts/pkg/types"
 	jsonpatch "gomodules.xyz/jsonpatch/v2"
+	coordinationv1 "k8s.io/api/coordination/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/utils/pointer"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	leasePrefix = "whereabouts"
+)
+
+func getLease(ctx context.Context, ipam *KubernetesIPAM, leaseName string) (*coordinationv1.Lease, error) {
+	lease := &coordinationv1.Lease{}
+	c := ipam.client
+	leaseTimeout := int32(ipam.config.LeaseDuration)
+
+	key := client.ObjectKey{
+		Name:      fmt.Sprintf("%s-%s", leasePrefix, leaseName),
+		Namespace: ipam.namespace}
+
+	err := c.Get(ctx, key, lease)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			logging.Errorf("failed to get lease: %s, error: %s", key.Name, err)
+			return nil, err
+		}
+		// create lease
+		lease = &coordinationv1.Lease{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-%s", leasePrefix, leaseName),
+				Namespace: ipam.namespace},
+			Spec: coordinationv1.LeaseSpec{
+				LeaseDurationSeconds: pointer.Int32Ptr(leaseTimeout),
```

Comment:
Is this really needed ?... Why don't you just `LeaseDurationSeconds: &leaseTimeout` ? 

Same for below; you can:
```golang
initialLeaseTransitions := uint32(0)
...
    Spec: coordinationv1.LeaseSpec{
        LeaseDurationSeconds: &leaseTimeout,
        LeaseTransitions:           &initialLeaseTransitions,
    },
...
```

I think this would remove the need for the `"k8s.io/utils/pointer"` dependency.

Reply:
It was not required. I have made the suggested change.